src/interfaces.ts

```diff
@@ -9,32 +9,50 @@ import { IForm, IValue } from './props';
 // much easier to call a function to fill in a partial definition
 // than to fill all this in (see further below for partials).
 interface IFieldConfigBase {
+  // Field is the ONLY required value of a fieldConfig
+  // All other attributes below are either optional or will be filled in
+  // by the component through the fillIn* functions
+  field: string;
+
+  // Another way of keeping fieldConfigs short are 'types' which are just
+  // different sets of defaults that can be quickly applied.
+  type: string;
+
+  // Core attributes
   className?: any;
-  colProps?: ColProps;
+  label: string | null;
+  nullify: boolean;
+  render: (...args: any[]) => string | JSX.Element | JSX.Element[];
```

Comment:
FYI, I noticed that Antd uses `React.ReactNode` and it works pretty well for anything if you don't need strict typing for children (which should be the case most of the time).
This should work: `render: (...args: any[]) => React.ReactNode;`
Except if I'm wrong :D

Reply:
Nice find. I have a TS PR coming right after this so I'll add that to that one, as it seems better grouped there.